### ISSUE TITLE
用deepspeed模型全量训练完后只保存的有chepoints的步数，外面的pytorch_model.bin只有928K,按说不应该是25G吗？

### ISSUE THREAD

**Issue by zhangyanbo2007** [LianjiaTech/BELLE#351]
![WXWorkCapture_16838682164669](https://github.com/LianjiaTech/BELLE/assets/9403745/f1da8c96-7eda-4763-8632-c0d2f6e0eb62)

**Comment by zhangyanbo2007** [comment]
补充一下，用的是最新的训练代码v2版本

**Comment by zhangyanbo2007** [comment]
#!/bin/bash
# Copyright (c) Microsoft Corporation.
# SPDX-License-Identifier: Apache-2.0
model_name_or_path=/root/.cache/huggingface/hub/LLaMA-13B # or LLaMA-13B
train_file=data/train_2M_CN_conv+alpaca_data_cleaned_conv_test_100.json
validation_file=data/train_2M_CN_conv+alpaca_data_cleaned_conv_test_100.json
output_dir=checkpoints/experiments_05112052/run_FT_13B_multi_42/output
cache_dir=checkpoints/experiments_05112052/run_FT_13B_multi_42/cache
mkdir -p ${output_dir}
mkdir -p ${cache_dir}
cutoff_len=1024

NCCL_DEBUG=INFO torchrun --nproc_per_node 8  --nnodes=3 --node_rank=0 --master_addr=192.168.68.104 src/train.py \
    --model_name_or_path ${model_name_or_path} \
    --llama \
    --deepspeed configs/deepspeed_config.json \
    --train_file ${train_file} \
    --validation_file ${validation_file} \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 2 \
    --gradient_accumulation_steps 4 \
    --num_train_epochs 2 \
    --model_max_length ${cutoff_len} \
    --save_strategy "steps" \
    --save_total_limit 3 \
    --learning_rate 8e-6 \
    --weight_decay 0.00001 \
    --warmup_ratio 0.05 \
    --lr_scheduler_type "cosine" \
    --logging_steps 10 \
    --evaluation_strategy "steps" \
    --fp16 True \
    --seed 1234 \
    --gradient_checkpointing True \
    --cache_dir ${cache_dir} \
    --output_dir ${output_dir}

**Comment by zhangyanbo2007** [comment]
是这个参数的问题吗？evaluation_strategy "steps"，意思是说只按保存的步数保存模型？

**Comment by barius** [comment]
@zhangyanbo2007 请把log文件发一下。
@xianghuisun 帮忙看下是否没存或者存失败了

**Comment by xianghuisun** [comment]
> ![WXWorkCapture_16838682164669](https://user-images.githubusercontent.com/9403745/237875556-f1da8c96-7eda-4763-8632-c0d2f6e0eb62.png)

已提供了多机训练的脚本，实测时没有出现你的问题

**Comment by zhangyanbo2007** [comment]
zero3模式出现的

**Comment by zhangyanbo2007** [comment]
我的zero3的checkpoint里面的zero_to_fp32.py为空
![WXWorkCapture_16839418289676](https://github.com/LianjiaTech/BELLE/assets/9403745/2b0e2591-65ba-4275-9cc3-1e3e29e95aa0)

**Comment by zhangyanbo2007** [comment]
我通过官网下载这个zero_to_fp32.py文件，是可以手动转换的，但这不是显得不智能嘛，看是不是哪个环节出问题了。

**Comment by Nicole-Yi** [comment]
同问，我也出现了这种状况，ZERO3，pytorch_model.bin只有1.2M
